msal/token_cache.py

```diff
@@ -172,7 +172,7 @@ def __add(self, event, now=None):
                     at["key_id"] = data.get("key_id")
                 self.modify(self.CredentialType.ACCESS_TOKEN, at, at)
 
-            if client_info:
+            if client_info and event.get("add_account") is not False:
```

Comment:
Is `is not False` necessary ?

Reply:
Unfortunately, it is currently necessary, because all other `acquire_token_by_xyz()` methods historically send no `add_account` flag at all, yet we want to treat such default case as "Yes we add an account".

That being said, you identified a bad judgement call I made here. Now I incline to change this flag's name to `skip_account`, so that we would be able to simply use `if client_info and not skip_account:` here.